File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        numsExpanded = [0]*2*n
        for i in range(0,len(numsExpanded)):
            numsExpanded[i] = nums[i%n]
        return numsExpanded
        

File: Data Structures & Algorithms/three-integer-sum/submission-0.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> result = new ArrayList<>();
        
        for(int i = 0; i < nums.length; i++) {
            int l=i+1;
            int r = nums.length-1;
            while(l < r) {
                if(nums[l]+nums[r] > -nums[i]) {
                    r--;
                } else if(nums[l] + nums[r] < -nums[i]){
                    l++;
                } else {
                    List<Integer> curr = new ArrayList<>();
                    curr.add(nums[l]);
                    curr.add(nums[r]);
                    curr.add(nums[i]);
                    if(!result.contains(curr)) {
                        result.add(curr);
                    }
                    l++;
                    r--;
                    
                }
            }

        }
        return result;
    }
}
